kingpin/actors/base.py

```diff
@@ -498,6 +499,181 @@ def execute(self):
+                      the options. You do not need to define it.*
+      :`_get_state`: Gets the current state of the resource.
+      :`_set_[option]`: A 'setter' for each option name passed in.
+      :`_get_[option]`: A 'getter' for each option name passed in.
+
+    **Optional Methods:**
+
+      :`_precache`: Called before any setters/getters are triggered. Used
+                       to optionally populate a cache of data to make the
+                       getters faster. For example, if you can make one API
+                       call to get all of the data about a resource, then
+                       store that data locally for fast access.
+
+      :`_compare_[option]`: Optionally you can write your own comparison
+                               method if you're not doing a pure string
+                               comparison between the source and destination.
+
+    **Examples**
+
+    .. code-block:: python
+
+        class MyClass(base.EnsurableBaseActor):
+
+            all_options = {
+                'name': (str, REQUIRED, 'Name of thing'),
+                'description': (str, None, 'Description of thing')
+            }
+
+            unmanaged_options = ['name']
+
+            @gen.coroutine
+            def _set_state(self):
+                if self.option('state') == 'absent':
+                    yield self.conn.delete_resource(
+                        name=self.option('name'))
+                else:
+                    yield self.conn.create_resource(
+                        name=self.option('name'),
+                        desc=self.option('description'))
+
+            @gen.coroutine
+            def _set_description(self):
+                yield self.conn.set_desc_of_resource(
+                    name=self.option('name'),
+                    desc=self.option('description'))
+
+            @gen.coroutine
+            def _get_description(self):
+                yield self.conn.get_desc_of_resource(
+                    name=self.option('name'))
+    """
+
+    # A list of option names that are _not_ automatically managed. These are
+    # useful if you have special behaviors like 'commit' on change, or if you
+    # have parameters that are unmutable ('name').
+    unmanaged_options = []
+
+    def __init__(self, *args, **kwargs):
+        # The 'state' parameter is a given, so make sure its set,
+        self.all_options['state'] = (
+            STATE, 'present', 'Desired state: present or absent')
+
+        # Now go ahead and validate all of the user inputs the normal way
+        super(EnsurableBaseActor, self).__init__(*args, **kwargs)
+
+        # Finally, do a class validation... make sure that we have actual
+        # getter/setter methods for each of the options. This populates dicts
+        # that provide references to the actual methods for execution later.
+        self._verify_methods()
+
+    def _verify_methods(self):
+        """Generates pointers to the Getter and Setter methods.
```

Comment:
Calling this method `verify` is misleading since it actually modifies the state of the object. I'd call it `gather` or `prepare`. Referring to them as pointers is wrong. You can say reference. Pointer has a very specific meaning